File: Modules/ServerTypes/Ticket/ProblemColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PriorityFormatter } from "../../Ticket/Priority/ColorPickerEditor";
import { ProblemRow } from "./ProblemRow";

export interface ProblemColumns {
    Id: Column<ProblemRow>;
    Name: Column<ProblemRow>;
    SystemName: Column<ProblemRow>;
    PriorityName: Column<ProblemRow>;
    PriorityColor: Column<ProblemRow>;
}

export class ProblemColumns extends ColumnsBase<ProblemRow> {
    static readonly columnsKey = 'Ticket.Problem';
    static readonly Fields = fieldsProxy<ProblemColumns>();
}

[PriorityFormatter]; // referenced types

File: Modules/Ticket/Problem/ProblemColumns.cs
namespace Ticketing.Ticket.Columns;

[ColumnsScript("Ticket.Problem")]
[BasedOnRow(typeof(ProblemRow), CheckNames = true)]
public class ProblemColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, SortOrder(1, false)]
    public int Id { get; set; }
    [EditLink, Width(500)]
    public string Name { get; set; }
    [Width(200), QuickFilter]
    public string SystemName { get; set; }
    [Width(200), QuickFilter]
    public string PriorityName { get; set; }
    [PriorityFormatter]
    public string PriorityColor { get; set; }
}